docs/proposals/DataTable/README.md

```diff
@@ -0,0 +1,165 @@
+# DataTable
+
+The `DataTable` Component would give the users options for pagination, sorting,
+and column visibility by leveraging some sort of API or service.
+
+## Design Patterns
+
+Tables are used to organize tabular information to help users understand and
+quickly scan for information.
+
+Some scenarios for Tables include the following:
+
+- Index Pages
+- Dashboards
+
+### Sorting
+
+Sorting can be enabled on column headers to improve usability.
+
+The default sorted column should be based on contextual based on user needs.
+
+To indicate the column being sorted, use an up (ascending) or down (descending)
+arrow next to the column title. Only 1 column can be sorted at a time. User can
+click on a sorted header to reverse the sorting order. Arrow will rotate 180
+degrees.
+
+Not all columns will need sorting enabled.
+
+# Column Visibility
+
+Not 100% sure if this is something to be implemented at component level but at
+least should be facilitated somehow.
+
+## Usage
+
+```tsx
+<DataTable
+  data={[
+    { name: "Gui", awesome: "yes" },
+    { name: "Harry", awesome: "yes" },
+    { name: "Chris", awesome: "yes" },
+    { name: "Juan", awesome: "yes" },
+  ]}
+  columns={[
+    {
+      accessorKey: "name",
+      cell: info => <span>{info.getValue()}</span>,
+    },
+    {
+      accessorKey: "awesome",
+      cell: info => info.getValue(),
+    },
+  ]}
+/>
+```
+
+## Interface and Props
+
+```ts
+interface DataTableProps<T> {
+  /**
+   * The actual data that will be used for the table.
+   * Typescript should infer T from typeof data.
+   *
+   * @type {T[]}
+   * @memberof DataTable
+   */
+  data: T[];
+  /**
+   * Should follow the @tanstack/react-table ColumnDef.
+   * I'm not sure if we need to create an abstraction around this
+   * specific type, ColumnDef by itself is already very flexible and powerfull enough
+   * to enable us to configure the table any way we like.
+   * https://tanstack.com/table/v8/docs/guide/column-defs
+   *
+   * @type {ColumnDef<T>[]}
+   * @memberof DataTable
+   */
+  columns: ColumnDef<T>[];
```

Comment:
so we basically have infinite flexibility here eh? That's awesome. Hopefully the amount of options do not make this over whelming. Part of me likes the idea of abstracting this to make it easier for developers to use but I don't want to limit them. Wondering if we can utilize the Column Helper in a way that makes things easier?
https://tanstack.com/table/v8/docs/guide/column-defs#column-helpers